### PR TITLE
Clarify contributing process

### PR DESCRIPTION
**Pull request by mattmcneeney**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,31 +1,37 @@
 # Contributing
 
-All contributors sending Pull Requests (PRs) must have a Contributor
+All contributors sending pull requests (PRs) must have a Contributor
 License Agreement on file as either an
 [individual](https://www.cloudfoundry.org/pdfs/CFF_Individual_CLA.pdf)
 or via their
 [employer](https://www.cloudfoundry.org/pdfs/CFF_Corporate_CLA.pdf).
 
+All new Github issues are PRs will be reviewed by the Project Management
```
Comment (duglin) [inline CONTRIBUTING.md]: s/are/and/  ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,60 +47,57 @@ potentially received objections/suggestions to be handled.
...
-specification changes.
-
-Note that not all issues will require a validation through implementation stage,
-and proposals can move back to a previous label at any time. When a PR is
-merged, the associated issue should be closed and any labels removed from the
-issue.
+The stages that a proposal will go through are as follows:
+- `1 - reviewing proposal`:
+  The API working group is actively reviewing a proposal that has been submitted
+  as a Github issue with the aim of validating both the problem statements and
```
Comment (duglin) [inline CONTRIBUTING.md]: isn't this process normally for PRs not issues?

Comment (mattmcneeney) [reply CONTRIBUTING.md]: We say that design changes have to have a google doc (via an issue)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -41,60 +47,57 @@ potentially received objections/suggestions to be handled.
...
+  any proposed solutions. Critical information such as new API endpoints and
+  response codes should be reviewed at this stage. Before moving to the next
+  stage, consensus must be agreed on the weekly call that the proposal is ready
+  to be validated.
+- `2 - validating through implementation`:
+  One of more platforms are actively working on the proposal with the aim of
```
Comment (jmrodri) [inline CONTRIBUTING.md]: Should this be "One or more"?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -41,60 +47,57 @@ potentially received objections/suggestions to be handled.
...
-[Open API document](openapi.yaml).
+be done by pointing to a branch in a Github repo with the proposed edits or by
+creating a PR with the `do not merge` label assigned to it. Reviewers, or
+potential implementers of the feature, can then easily see the exact changes
+being proposed so that they can comment on them. If the proposal adds or changes
+an object model or resource endpoint definition, it is expected that the PR will
+also include the necessary updates to the [Open API document](openapi.yaml) and
+[Swagger document](swagger.yaml).
 
 Once support for the change has been implemented, a PR should be sent to this
```
Comment (duglin) [inline CONTRIBUTING.md]: "has been implemented" - by who/how? We don't talk about validation until later so this feels like we're missing something in the flow.

File: CONTRIBUTING.md (hunk 3 above)
Comment (duglin) [reply CONTRIBUTING.md]: +1

Comment (mattmcneeney) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (mattmcneeney) [reply CONTRIBUTING.md]: Good point. I've combined a couple of sections to split into just "Minor Changes" and "Major Changes". Let me know what you think...

**Comment by duglin** [comment]
2 more reviews needed

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,100 +1,93 @@
...
-Note that not all issues will require a validation through implementation stage,
-and proposals can move back to a previous label at any time. When a PR is
-merged, the associated issue should be closed and any labels removed from the
-issue.
+as on our weekly calls. Once the working group have agreed to focus on a
+proposal, it will move into the first stage, as described below.
+
+The stages that a proposal will go through are as follows:
+- `1 - reviewing proposal`:
+  The API working group is actively reviewing a proposal that has been submitted
```
Comment (pmorie) [inline CONTRIBUTING.md]: strictly non-technical, this is quite a wall of text - consider splitting this up into smaller paragraphs?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,100 +1,93 @@
...
+  as a Github issue with the aim of validating both the problem statements and
+  any proposed solutions. Critical information such as new API endpoints and
+  response codes should be reviewed at this stage. For changes largely impacting
+  the Open Service Broker API actors (Platforms, Service Authors, etc), it is
+  recommended to solicit feedback from these actors and leave enough time (say
+  2 weeks) for feedback to be provided, and for the potentially received
+  objections/suggestions to be handled.
+  Once the design has been finalized in the Google Doc, the proposed set of
```
Comment (pmorie) [inline CONTRIBUTING.md]: One thing I think we should note here that comes from the F2F - it might be helpful to mention that small, focused changes are easier to make progress on.

**Comment by duglin** [comment]
@mattmcneeney did you want to address the comments on this?

**Comment by duglin** [comment]
@mattmcneeney did you want to address the comments on this?

**Comment by mattmcneeney** [comment]
@duglin I think the only actionable thing was to break up a wall of text which I think we could do later / if we get feedback that the guide is hard to follow. WDYT?

**Comment by duglin** [comment]
Needs 2 more reviews - I think we can deal with the "split" in a follow-on PR if needed.
ping @fmui @kibbles-n-bytes @vaikas-google @zrob

**Comment by zrob** [comment]
lgtm

**Comment by fmui** [comment]
LGTM
